nav2_map_server/CMakeLists.txt

```diff
@@ -43,26 +63,47 @@ set(map_io_library_name map_io)
 
 set(library_name ${map_server_executable}_core)
 
-add_library(${map_io_library_name} SHARED
-  src/map_mode.cpp
-  src/map_io.cpp)
+add_library(${map_io_library_name}_2d SHARED
+  src/map_2d/map_mode.cpp
+  src/map_2d/map_io_2d.cpp)
 
-add_library(${library_name} SHARED
-  src/map_server/map_server.cpp
-  src/map_saver/map_saver.cpp
-  src/costmap_filter_info/costmap_filter_info_server.cpp)
+add_library(${map_io_library_name}_3d SHARED
+  src/map_3d/pcl_helper.cpp
+  src/map_3d/map_io_3d.cpp)
 
-set(map_io_dependencies
+add_library(${library_name} SHARED
+  src/costmap_filter_info/costmap_filter_info_server.cpp
+  include/nav2_map_server/map_server.hpp
+  include/nav2_map_server/map_saver.hpp
```

Comment:
These files act as a joint b/w all of the different `map-server` implementations, although these are not needed by the library itself, they act as an interface. Since we are using the `include_directories()` utility of CMake, we don't need that.

Reply:
Ok, it's working now, marking this as resolved.